servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ClientEffectiveStrategyTest.java

```diff
@@ -44,277 +43,284 @@
-                true, false, true, false,
-                () -> userStrategy(customStrategyBuilder().offloadAll().build()))),
-        customUserStrategyWithCF(clientType -> new Params(clientType, Offloads.ALL,
-                true, false, false, true,
-                () -> userStrategy(customStrategyBuilder().offloadAll().build())));
-
-        private final Function<ClientType, Params> paramsProvider;
-
-        ClientStrategyCase(Function<ClientType, Params> paramsProvider) {
-            this.paramsProvider = paramsProvider;
-        }
-
-        @Override
-        public Params apply(ClientType clientType) {
-            return paramsProvider.apply(clientType);
-        }
-    }
-
-    @ParameterizedTest
-    @EnumSource(ClientStrategyCase.class)
-    void blocking(ClientStrategyCase strategyCase) throws Exception {
-        try (Params params = strategyCase.apply(ClientType.Blocking)) {
-            assertThat("Null params supplied", params, notNullValue());
-            BlockingHttpClient blockingClient = params.client().asBlockingClient();
-            HttpResponse response = blockingClient.request(blockingClient.get("/"));
-            assertThat(response.payloadBody().toString(StandardCharsets.US_ASCII), is(GREETING));
-            params.verifyOffloads();
+    private static final HttpExecutionStrategy[] BUILDER_STRATEGIES = {
+            null, // unspecified
+            noOffloadsStrategy(),
+            anyStrategy(),
+            defaultStrategy(),
+            HttpExecutionStrategies.customStrategyBuilder().offloadSend().build(),
+            offloadAll(),
+    };
+
+    private static final HttpExecutionStrategy[] FILTER_STRATEGIES = {
+            null, // absent
+            noOffloadsStrategy(), // treated as "anyStrategy"
+            anyStrategy(),
+            defaultStrategy(), // treated as "anyStrategy"
+            HttpExecutionStrategies.customStrategyBuilder().offloadSend().build(),
+            offloadAll(),
+    };
+
+    private static final HttpExecutionStrategy[] LB_STRATEGIES = {
+            null, // absent
+            noOffloadsStrategy(), // treated as "anyStrategy"
+            anyStrategy(),
+            defaultStrategy(), // treated as "anyStrategy"
+            HttpExecutionStrategies.customStrategyBuilder().offloadSend().build(),
+            offloadAll(),
+    };
+
+    private static final HttpExecutionStrategy[] CF_STRATEGIES = {
+            null, // absent
+            noOffloadsStrategy(), // treated as "anyStrategy"
+            anyStrategy(),
+            defaultStrategy(), // treated as "anyStrategy"
+            HttpExecutionStrategies.customStrategyBuilder().offloadSend().build(),
+            offloadAll(),
+    };
+
+    static final ServerContext context;
+
+    static {
+        try {
+            HttpServerBuilder serverBuilder = HttpServers.forAddress(localAddress(0));
+            context = serverBuilder.listenBlocking((ctx, request, responseFactory) ->
```

Comment:
should this be created in a `@BeforeAll` since it is cleaned up in a `@AfterAll`?

Reply:
Using `@BeforeAll` would require that the static be mutable. I (we?, everyone?) prefer `static  final` when possible so we end up with a wart of having to use `@AfterAll` without before all. Making a `ServerContext` extension as we have done with `ExecutionContextExtension` and `ExecutorExtension` would be the next step to take if we use `ServerContext` in this way in multiple tests.